File: Client/src/api/UserCompany.js
import api from "./api.js";

export const fetchMyCompanies = async () => {
    const response = await api.get("UserCompany/my-companies");
    return response.data || [];
  };

export const switchCompany = async (companyId) => {
    const response = await api.post(
      `UserCompany/switch-company?companyId=${companyId}`,
    );
    return response.data || {};
  }
;

File: Server/HRConnect.Api/Models/UserCompany.cs
namespace HRConnect.Api.Models
{
    using System;
    using System.Collections.Generic;
    using HRConnect.Api.Models;
    using System.Linq;
    using System.Threading.Tasks;
    public class UserCompany
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public string CompanyId { get; set; } = string.Empty;
        public Company Company { get; set; } = null!;
        public bool IsDefault { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    }
}